pulsar-common/src/main/java/org/apache/pulsar/common/protocol/Commands.java

```diff
@@ -801,6 +805,20 @@ private static void convertSchema(SchemaInfo schemaInfo, Schema schema) {
         });
     }
 
+    private static void convertAutoConsumeSchema(SchemaInfo schemaInfo, Schema schema) {
+        schema.setName(schemaInfo.getName())
+                .setSchemaData(schemaInfo.getSchema())
+                .setType(Schema.Type.AutoConsume);
```

Comment:
repeat with 'convertSchema(SchemaInfo schemaInfo, Schema schema)' except this line, so can add a method `private convertSchemaWithoutSchemaType(SchemaInfo schemaInfo, Schema schema)`

Reply:
Here, `setType` is set to `Schema.type.AutoConsume` because the value of AutoConsume in `Schema.Type` is 21 and the value of AutoConsume in `SchemaType` is -3. It's an act of convert the AutoConsume schema, so when you say "without SchemaType", it doesn't seem to mean AutoConsume schema alone. What do you think of my thoughts here.

Reply:
Please reduce duplicated code between `convertSchema` and `convertAutoConsumeSchema`. Don't just copy and paste. We should reuse the existing logic rather than copy a method to another method and just modify one line.

A simple solution is to modify the `getSchemaType` method like:

```java
    private static Schema.Type getSchemaType(SchemaType type) {
        if (type.equals(SchemaType.AUTO_CONSUME)) {
            return Schema.Type.AutoConsume;
        } else if (type.getValue() < 0) {
            return Schema.Type.None;
```

A more simple solution is just changing the `SchemaType.AUTO_CONSUME` to 21 so that the value could be consistent with what's defined in the `PulsarApi.proto`.

Reply:
How about we do it this way? 
```java
    private static void convertAutoConsumeSchema(SchemaInfo schemaInfo, Schema schema) {
        convertSchema(schemaInfo, schema);
        schema.setType(Schema.Type.AutoConsume);
    }
```
`getSchemaType` is a public method. And `SchemaType` is also public. It's better not to change.

Reply:
The one that accepts a `SchemaType` is private, not public.

Regarding the public one, I agree it's better not to change and keep the code in `ServerCnx` not changed.

Reply:
https://github.com/apache/pulsar/blob/fa6af432ef3d015b371121afd9324ba7f393994d/pulsar-client-api/src/main/java/org/apache/pulsar/common/schema/SchemaType.java#L178-L180
If we change the value of `AUTO_CONSUME` in `SchemaType` to 21, the behavior may change in two places: 
https://github.com/apache/pulsar/blob/fa6af432ef3d015b371121afd9324ba7f393994d/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java#L758-L761
https://github.com/apache/pulsar/blob/fa6af432ef3d015b371121afd9324ba7f393994d/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java#L613-L619

So how about we make that change
```java
    private static void convertAutoConsumeSchema(SchemaInfo schemaInfo, Schema schema) {
        convertSchema(schemaInfo, schema);
        schema.setType(Schema.Type.AutoConsume);
    }
```
And in `ServerCnx`.
```java
            .type(protocolSchema.getType() == Schema.Type.AutoConsume
                    ? SchemaType.AUTO_CONSUME : Commands.getSchemaType(protocolSchema.getType()))
```

Reply:
I mean, you can just modify this method:

https://github.com/apache/pulsar/blob/fa6af432ef3d015b371121afd9324ba7f393994d/pulsar-common/src/main/java/org/apache/pulsar/common/protocol/Commands.java#L773

There is no need to add a new method to do that.

Reply:
I didn't catch your meaning just now, sorry. It's been fixed.